modules/oadp-ssec-encrypted-backups.adoc

```diff
@@ -0,0 +1,143 @@
+
+[source,yaml]
+----
+ snapshotLocations:
+  - velero:
+      config:
+        profile: default
+        region: <region>
+      provider: aws
+# ...
+----
+.Procedure
+. Create an SSE-C encryption key by running the following command:
++
+[source,terminal]
+----
+$ dd if=/dev/urandom bs=1 count=32 > sse.key
+cat sse.key | base64 > sse_encoded.key
+ln -s sse_encoded.key customer-key
+----
+. Create an OpenShift secret.
+.. If you are initially installing and configuring {oadp-short}, you can create the AWS credential and encryption key secret at the same time by running the following command:
++
+[source,terminal]
+----
+$ oc create secret generic cloud-credentials --namespace openshift-adp --from-file cloud=<path>/openshift_aws_credentials,customer-key=<path>/sse_encoded.key
+----
+.. If you are updating an existing installation, edit the values of the `cloud-credential` `secret` block of the `DataProtectionApplication` CR manifest, as in the following example:
++
+[source,yaml]
+----
+apiVersion: v1
+data:
+  cloud: W2Rfa2V5X2lkPSJBS0lBVkJRWUIyRkQ0TlFHRFFPQiIKYXdzX3NlY3JldF9hY2Nlc3Nfa2V5P<snip>rUE1mNWVSbTN5K2FpeWhUTUQyQk1WZHBOIgo=
+  customer-key: v+<snip>TFIiq6aaXPbj8dhos=
+kind: Secret
+# ...
+----
+. Edit the value of the `customerKeyEncryptionFile` attribute in the `backupLocations` block of the `DataProtectionApplication` CR manifest, as in the following example:
++
+[source,yaml]
+----
+spec:
+  backupLocations:
+    - velero:
+        config:
+          customerKeyEncryptionFile: /credentials/customer-key
+          profile: default
+# ...
+----
++
+[WARNING]
+====
+Existing installations require you to restart the Velero pod to remount the secret credentials properly.
+====
++
+The installation is complete, and you can back up and restore OpenShift resources. The data saved in AWS S3 storage is encrypted with the new key, and you cannot download it from the AWS S3 console or API without the additional encryption key.
+
+.Verification
+The purpose of this test is to verify that you cannot download the encrypted files without the inclusion of an additional key.
+
+. Create a test file by running the following command:
++
+[source,terminal]
+----
+$ echo "encrypt me please" > test.txt
+----
+. Upload the test file by running the following command:
++
+[source,terminal]
+----
+$ aws s3api put-object \
+  --bucket <bucket> \
+  --key test.txt \
+  --body test.txt \
+  --sse-customer-key fileb://sse.key \
+  --sse-customer-algorithm AES256
+----
+. Try to download the file, this should fail because the file is now encrypted with an additional key. In either the Amazon web console or the terminal, run the following command:
++
+[source,terminal]
+----
+$ s3cmd get s3://<bucket>/test.txt test.txt
+----
+. Download the file with the additional encryption key by running the following commands:
++
```

Comment:
@anarnold97 OCP is almost sure to ask me to make each code block a separate step, but I can't figure out a lead-in for the second command. The source material also treats these two commands as a single step. https://hackmd.io/nWTqKj4dRxiU3PCuAKtYuQ  Thoughts?